server/service/osquery_utils/queries.go

```diff
@@ -1187,17 +1187,25 @@ func GetDetailQueries(fleetConfig config.FleetConfig, features *fleet.Features)
 		generatedMap["scheduled_query_stats"] = scheduledQueryStats
 	}
 
-	for _, env := range os.Environ() {
-		prefix := "FLEET_DANGEROUS_REPLACE_"
-		if !strings.HasPrefix(env, prefix) {
-			continue
+	if features != nil {
+		var unknownQueries []string
+
+		for name, override := range features.DetailQueryOverrides {
+			if override == nil {
+				delete(generatedMap, name)
```

Comment:
What about unknown queries to be deleted? (You should do the `query, ok` check before this)

Reply:
Good point.